# Exploratory Data Analysis

In [2]:
import pandas as pd
import numpy as np

In [6]:
df_2000 = pd.read_csv("Data/final_tmdb_data_2000.csv.gz", low_memory = False)

In [7]:
df_2001 = pd.read_csv("Data/final_tmdb_data_2001.csv.gz", low_memory = False)

In [10]:
EDA_df = pd.concat([df_2000, df_2001])
EDA_df.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]      NaN   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...      NaN   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   

         id original_language   original_title  ...     revenue  runtime  \
0       NaN               NaN              NaN  ...         NaN      NaN   
1   62127.0                en  The Fantasticks  ...         0.0     86.0   
2  110977.0                en    For the Cause  ...         0.0    100.0   
3  442869.0                hi             Gang  ...         0.0    152.0   
4     843.0                cn             花樣年華  ...  12854953.0     99.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         5.500       22.0           NaN  
2    0.0         5.100        8.0           NaN  
3    0.0         4.000        1.0           NaN  
4    0.0         8.113     2101.0            PG  

[5 rows x 26 columns]

In [32]:
EDA_df.to_csv("Data/tmdb_results_combined.csv.gz",compression='gzip',index=False)

1. How many movies had at least some valid financial information (values > 0 for budget OR revenue)?

In [18]:
budgets = (EDA_df['budget'] > 0)
budgets.value_counts()

False    1977
True      540
Name: budget, dtype: int64

In [19]:
EDA_df = EDA_df.loc[budgets]

In [20]:
revenues = (EDA_df['revenue'] > 0)
revenues.value_counts()

True     352
False    188
Name: revenue, dtype: int64

In [21]:
EDA_df = EDA_df.loc[revenues]

- Answer: 540 movies had budget information listed and 352 movies had revenue information listed.

2. How many movies are there in each of the certification categories (G/PG/PG-13/R)?

In [25]:
EDA_df['certification'].value_counts()

R        142
PG-13    117
PG        30
G         13
NR         6
Name: certification, dtype: int64

3. What is the average revenue per certification category?


In [36]:
EDA_df.groupby('certification', as_index=False)['revenue'].mean()

certification       revenue
0             G  1.332169e+08
1            NR  2.382332e+07
2            PG  1.304904e+08
3         PG-13  1.110740e+08
4             R  5.294547e+07

4. What is the average budget per certification category?

In [37]:
EDA_df.groupby('certification', as_index=False)['budget'].mean()

certification        budget
0             G  4.400000e+07
1            NR  1.478333e+07
2            PG  5.218333e+07
3         PG-13  4.680682e+07
4             R  2.706348e+07